### ISSUE TITLE
Deletion of snapshots when `.` in test file name

### ISSUE THREAD

**Issue by ms609** [r-lib/testthat#1425]
When a test file's name contains a period, snapshots are deleted after testing.

To reproduce, create a file `test-my.func.R` containing
```r
test_that("test-cases-saved", {
  vdiffr::expect_doppelganger("simple-plot", {
    plot(1, 1)
  })
})
```

`devtools::test()` or `testthat::test_file()` deliver output:

```
Deleting unused snapshots:
* _snaps/my.func
```

After renaming the file to remove the period, the test files are retained as expected.

**Comment by hadley** [comment]
Thanks for filing this bug report! Unfortunately because we think it's unlikely to affect many people and there's an easy work around (don't use multiple extensions on test files), we don't have the development resources to fix at this time. It's our policy to close such issues to help stay focussed on the biggest problems, but the issue is still indexed by google, so if other people hit it, they'll be able to find it, and we can consider reopen it if it turns out to be a common problem. Thanks for reporting and I'm sorry we couldn't help more 😞.

**Comment by ms609** [comment]
Thanks; I completely understand the need to concentrate developer effort on the big fish!

If I were to take a look at proposing a fix myself, would this be appreciated? I get that user-submitted patches can sometimes take a fair bit of work to evaluate and maintain too, so maybe this wouldn't be the best use of time?

**Comment by hadley** [comment]
Unfortunately it's hard to say — it'd be appreciated if the fix is pretty simple (e.g. because I messed up something simple with how extensions are handled) but not if it requires a bunch of changes. I would be less interested if it was a complex fix as the way all the snapshot file handling stuff comes together is already hard for me to understand, so I wouldn't want to make it any harder.

**Comment by ms609** [comment]
A simple fix is proposed in #1441.
